### PR TITLE
[MySQL] `az mysql flexible-server create/update/restore`: Add `database-port` and `maintenance-policy-patch-strategy` params

### PR DESCRIPTION
**Pull request by honghr**
**Related command**

az mysql flexible-server create/update: add --maintenance-policy-patch-strategy
az mysql flexible-server create/restore: add --database-port
az mysql flexible-server replica create: add --database-port

**Description**

Add new param database-port to support customize database port for MySQL vnet server.
Add new param maintenance-policy-patch-strategy to support customize patch strategy and accepted values are Regular and VirtualCanary.

**History Notes**

[Mysql] `az mysql flexible-server create/update`: Add `--maintenance-policy-patch-strategy`
[Mysql] `az mysql flexible-server create/restore`: Add `--database-port`
[Mysql] `az mysql flexible-server replica create`: Add `--database-port`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/azure-cli/azure/cli/command_modules/mysql/_util.py [hunk 1]
```diff
@@ -374,19 +373,12 @@ def _resolve_api_version(client, provider_namespace, resource_type, parent_path)
 
 def run_subprocess(command, stdout_show=None):
     if stdout_show:
-        process = subprocess.Popen(command, shell=True)
+        process = run_cmd(command)
     else:
-        process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        process = run_cmd(command, capture_output=True)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/mysql/_util.py:378]: cc @AllyW for review

**Review by evelyn-ys — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/mysql/_util.py [hunk 2]
```diff
@@ -374,19 +373,12 @@ def _resolve_api_version(client, provider_namespace, resource_type, parent_path)
...
     if process.returncode:
-        logger.warning(process.stderr.read().strip().decode('UTF-8'))
-
-
-def run_subprocess_get_output(command):
-    commands = command.split()
-    process = subprocess.Popen(commands, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    process.wait()
-    return process
+        logger.warning(process.stderr.strip().decode('UTF-8'))
```
Comment (AllyW) [inline src/azure-cli/azure/cli/command_modules/mysql/_util.py:381]:
```suggestion
        if stdout_show:
          logger.warning("Error in running cmd {}".format("|".join(command)))
        else:
          logger.warning(process.stderr.strip().decode('UTF-8'))
```

Comment (AllyW) [reply src/azure-cli/azure/cli/command_modules/mysql/_util.py]: When `capture_output=False`, stderr would be printed and `process.stderr` would be `None`

File: src/azure-cli/azure/cli/command_modules/mysql/_util.py [hunk 3]
```diff
@@ -374,19 +373,14 @@ def _resolve_api_version(client, provider_namespace, resource_type, parent_path)
...
-        logger.warning(process.stderr.read().strip().decode('UTF-8'))
-
-
-def run_subprocess_get_output(command):
-    commands = command.split()
-    process = subprocess.Popen(commands, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    process.wait()
-    return process
+        if stdout_show:
+            logger.warning("Error in running cmd | %s", command)
```
Comment (AllyW) [inline src/azure-cli/azure/cli/command_modules/mysql/_util.py:381]: `logger.warning("Error in running cmd %s", "|".join(command))` ?

**Review by AllyW — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**